src/github_utils.py

```diff
@@ -19,48 +19,56 @@ def get_pull_request_diff():
     # Get diff content
     diff_url = pull_request.diff_url
     response = requests.get(diff_url)
-    print(response.text)
+    #print(response.text)
     return response.text
 
 
-def get_valid_lines(diff_content):
-    """Parse diff to find valid line numbers with proper initialization"""
-    valid_lines = {}
+def get_valid_hunks(diff_content):
+    """Parse diff to find valid hunk positions"""
+    hunks = []
+    current_hunk = None
     current_file = None
-    line_number = None
```

Comment:
The variable `hunk_length` isn't validated for correct format before converting into int, this can lead to unexpected program termination.

```suggestion
Add more error checking when parsing the hunk header, especially for `new_start` and `hunk_length`. Use a regular expression for more robust parsing. For example:
``````python
import re
match = re.match(r'@@ -[0-9]+,[0-9]+ \+([0-9]+),([0-9]+) @@', line)
if match:
new_start = int(match.group(1))
hunk_length = int(match.group(2))
...
else:
print(f"Invalid hunk header: {line}")
continue # Or handle the error as appropriate
```